monkey/tests/unit_tests/infection_monkey/utils/test_timer.py

```diff
@@ -0,0 +1,69 @@
+import time
+
+import pytest
+
+from infection_monkey.utils.timer import Timer
+
+
+@pytest.fixture
+def start_time(set_current_time):
+    start_time = 100
+    set_current_time(start_time)
+
+    return start_time
+
+
+@pytest.fixture
+def set_current_time(monkeypatch):
+    def inner(current_time):
+        monkeypatch.setattr(time, "time", lambda: current_time)
+
+    return inner
+
+
+@pytest.mark.parametrize(("timeout"), [5, 1.25])
+def test_timer_not_expired(start_time, set_current_time, timeout):
+    t = Timer()
+    t.set(timeout)
+
+    assert not t.is_expired()
+
+    set_current_time(start_time + (timeout - 0.001))
+    assert not t.is_expired()
```

Comment:
The first is testing that the timer is not expired when `current_time == start_time`. The second test tests a time between the start time and the expected expiration time.

Reply:
Maybe we should separate this into two functions since they're testing two different cases.

Reply:
Since the nature of the object is that its state changes overtime, I'd prefer to leave them as is. The tests are easier to understand: advance the clock, check that the state is correct, repeat. If it's split into multiple tests, I think it loses some continuity since you need to restart the clock.